pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -0,0 +1,647 @@
+			// We found an unfinished job that has us as the parent, but it is not in our Active list.
+			// This could happen if we crashed right after creating the Job and before updating the status,
+			// or if our jobs list is newer than our cj status after a relist, or if someone intentionally created
+			// a job that they wanted us to adopt.
+
+			// TODO: maybe handle the adoption case?  Concurrency/suspend rules will not apply in that case, obviously, since we can't
+			// stop users from creating jobs if they have permission.  It is assumed that if a
+			// user has permission to create a job within a namespace, then they have permission to make any cronJob
+			// in the same namespace "adopt" that job.  ReplicaSets and their Pods work the same way.
+			// TBS: how to update cj.Status.LastScheduleTime if the adopted job is newer than any we knew about?
+		} else if found && IsJobFinished(&j) {
+			_, status := getFinishedStatus(&j)
+			deleteFromActiveList(cj, j.ObjectMeta.UID)
+			recorder.Eventf(cj, corev1.EventTypeNormal, "SawCompletedJob", "Saw completed job: %s, status: %v", j.Name, status)
+		}
+	}
+
+	// Remove any job reference from the active list if the corresponding job does not exist any more.
+	// Otherwise, the cronjob may be stuck in active mode forever even though there is no matching
+	// job running.
+	for _, j := range cj.Status.Active {
+		if found := childrenJobs[j.UID]; !found {
+			recorder.Eventf(cj, corev1.EventTypeNormal, "MissingJob", "Active job went missing: %v", j.Name)
+			deleteFromActiveList(cj, j.UID)
+		}
+	}
+
+	// TODO: @alpatel explore if cached client can be used as realCJControl
+	updatedCJ, err := cjc.UpdateStatus(cj)
+	if err != nil {
+		klog.V(2).InfoS("Unable to update status for cronjon", "cronjob", klog.KRef(cj.GetNamespace(), cj.GetName()), "resourceVersion", cj.ResourceVersion, "err", err)
+		return nil, err
+	}
+	*cj = *updatedCJ
+
+	if cj.DeletionTimestamp != nil {
+		// The CronJob is being deleted.
+		// Don't do anything other than updating status.
+		return nil, fmt.Errorf("cronjob %s/%s is being deleted", cj.Namespace, cj.Name)
+	}
+
+	if cj.Spec.Suspend != nil && *cj.Spec.Suspend {
+		klog.V(4).InfoS("Not starting job because the cron is suspended", "cronjob", klog.KRef(cj.GetNamespace(), cj.GetName()))
+		return nil, nil
+	}
+
+	// TODO: alpatel: this is here now just to pass the unit tests,
+	//  move this to the start of the function to return on error early
+	sched, err := cron.ParseStandard(cj.Spec.Schedule)
+	if err != nil {
+		// this is likely a user error in defining the spec value
+		// we should log the error and not reconcile this cronjob until an update to spec
+		klog.V(2).InfoS("unparseable schedule", "cronjob", klog.KRef(cj.GetNamespace(), cj.GetName()), "schedule", cj.Spec.Schedule, "err", err)
+		recorder.Eventf(cj, corev1.EventTypeWarning, "UnparseableSchedule", "unparseable schedule: %s : %s", cj.Spec.Schedule, err)
+		return nil, nil
```

Comment:
add a comment explaining that you aren't returning an error because requeueing would simply produce the same parse error over and over until they update, which will cause a requeue in an of itself.

Reply:
comments are present at the start of the `if` block can move them here if needed. 

https://github.com/kubernetes/kubernetes/pull/93370/files#diff-3ac32b4c4bbfd547f2224dda02e1b422629791f723459a6cfb6a1a0670519975R372